src/ir/item.rs

```diff
@@ -1443,10 +1457,18 @@ impl ItemCanonicalPath for Item {
     fn namespace_aware_canonical_path(&self,
                                       ctx: &BindgenContext)
                                       -> Vec<String> {
-        let path = self.canonical_path(ctx);
+        let mut path = self.canonical_path(ctx);
+        let is_constified_module_enum = self.is_constified_enum_module(ctx);
         if ctx.options().enable_cxx_namespaces {
+            if is_constified_module_enum {
+                path.push(CONSTIFIED_ENUM_MODULE_REPR_NAME.into());
+            }
             return path;
         }
+        if is_constified_module_enum {
```

Comment:
To fix the last test-case I commented about, you should remove this early return at let it arrive to the last return.

For that to work, you need to add the `REPR_NAME` to the path, so this method could be structured like:

```
if is_constified_module_enum {
    path.push(CONSTIFIED_ENUM_MODULE_REPR_NAME.into());
}

if ctx.options().enable_cxx_namespaces {
    return path;
}

if ctx.options().disable_name_namespacing {
    let trailing_segments = if is_constified_module_enum { 2 } else { 1 };
    return path[..path.len() - trailing_segments].iter().cloned().collect();
}

return vec![path[1..].join("_")];
```